### DISCUSSION TITLE
How to check key for value before revalidate

### DISCUSSION THREAD

**Discussion by daveteu** [vercel/swr#675] [Q&A]
currently in page I use this

`const { data, error, mutate } = useSWR('/api/', fetcher)`

in child component I use this so I can get the values

`const { data, error, mutate } useSWR('/api', fetcher, { revalidateOnMount: false}`

Is there anyway I can access the current key value, check if the key is "fetched before" before fetching? 

**pseudo code**
`const { data, error, mutate } = useSWR('/api', (currentkeydata) => fetcher(check if currentkeydata exist before fetch))`

**Comment by sergiodxa** [comment]
You can import the internal cache doing:
```js
import { cache } from "swr";
```
And then read a value using `cache.get(key)`.

However, I'm not sure what do you plan to achieve, do you want to effectively prevent SWR from revalidating your data? If so, why use SWR?
